src/services/CertificateGenerationService.ts

```diff
@@ -1381,6 +1382,24 @@ class CertificateGenerationService {
       )
     );
   }
+
+  /**
+   * Sorts required required standards if present firstly by sectionNumber and then by rsNumber and then returns it
+   * @param requiredStandards - the requiredStandards collection to sort
+   * @returns - the sorted requiredStandards collection
+   */
+  private sortRequiredStandards = (requiredStandards: IRequiredStandard[] | undefined): IRequiredStandard[] | undefined => {
+    if (!requiredStandards) {
+      return;
```

Comment:
Not sure about returning undefined here, may be better to return an empty array

Reply:
Could move the check so the sort method is only called if RS is defined?

Reply:
Kristian is right, although theoretically the test result would not have been sent to cert-gen if the requiredStandards array was empty or undefined due to this logic in the cert-gen-init function:

`const hasNonEmptyRequiredStandards = !!requiredStandards?.length;`